### PR TITLE
feat: Greetings.yml added

### LINKED ISSUE #33: [Feature]: Add the github action workflow for greeting contributors
Is your feature request related to a problem? Please describe.
The repo needs a greeting workflow

Describe the solution you'd like
The current project lacks an automated action to greet contributors. GitHub Actions can significantly enhance project automation. I propose adding a GitHub Action workflow to greet new pull request (PR) and issue creators, fostering a welcoming environment.

Describe alternatives you've considered
Implement a GitHub Action workflow that triggers on new PRs and issues. The workflow should include a greeting message to acknowledge and appreciate the contributors for their efforts.

### PR DESCRIPTION
**Pull request by praptisharma28**
closes #33

### CODE REVIEW

**Comment by tushar5526** [comment]
@praptisharma28 you will have to install pre-commit `pip install pre-commit` and `pre-commit install` and then run pre-commit on all files to fix the error. `pre-commit run --all-files`

**Comment by praptisharma28** [comment]
Added precomit in the environment variables , still its not recognised ? @tushar5526

**Comment by tushar5526** [comment]
https://pre-commit.com/ - Setup this

**Comment by praptisharma28** [comment]
done @tushar5526

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,5 @@ pyjwt==2.8.0
 Flask-HTTPAuth==4.8.0
 python-dotenv==1.0.0
 requests==2.31.0
-filelock==3.13.1
\ No newline at end of file
+filelock==3.13.1
+pre-commit
```
Comment (tushar5526) [inline requirements.txt:8]: pre-commit will rather go in `requirements-dev.txt` as it is a dev requirement not a core requirement.

**Review by tushar5526 — CHANGES_REQUESTED** [review]
Move pre-commit to dev requirements.

**Comment by praptisharma28** [comment]
ok done @tushar5526 , dev already have a version of pre-commit. removed from requirements.txt . You can check now.

**Comment by praptisharma28** [comment]
@tushar5526 do I need to do anything more?

**Pull request merged**